subsystems/drivebase/SwerveDrive.java

```diff
@@ -288,28 +330,36 @@ public void initCoastMode() {
 
     public void toggleIdleMode() {
         for (SwerveModule m : mModules) {
-            if (m.getDriveMotor().getIdleMode() == IdleMode.kBrake)
-                    m.getDriveMotor().setIdleMode(IdleMode.kCoast);
+            if (((CANSparkMax) m.getDriveMotor()).getIdleMode() != IdleMode.kBrake)
+                    ((CANSparkMax) m.getDriveMotor()).setIdleMode(IdleMode.kCoast);
             else {
-                m.getDriveMotor().setIdleMode(IdleMode.kBrake);}
+                ((CANSparkMax) m.getDriveMotor()).setIdleMode(IdleMode.kBrake);}
             //We do not want to toggle steer motor idle mode 
         }
     }
 
     public void setMotorIdleMode(IdleMode mode) {
         for (SwerveModule m : mModules) {
-            (m.getDriveMotor()).setIdleMode(mode);
+            ((CANSparkMax) m.getDriveMotor()).setIdleMode(mode);
         }
     }
 
+    public void driveRobotRelative(ChassisSpeeds speeds) {
+        SwerveModuleState[] targetStates = kinematics.toSwerveModuleStates(speeds);
+    
```

Comment:
I'm not going to keep adding the "why are we changing this" comments in this file. Just assume that for all this stuff that has nothing to do with advantagekit